test/e2e-common.sh

```diff
@@ -210,6 +211,78 @@ function enable_monitoring(){
       --role roles/cloudtrace.agent
 }
 
+function test_authentication_check_for_brokercell() {
+  echo "Starting authentication check test for brokercell."
+  local auth_mode=${1}
+  ko apply -f ${CONFIG_WARMUP_GCP_BROKER}
+
+  echo "Starting authentication check test which is running outside of the broker related Pods."
+  wait_until_brokercell_authentication_check_pending "$(non_pod_check_keywords)" || return 1
+
+  echo "Starting authentication check test which is running inside of the broker related Pods."
+  apply_invalid_resource "$auth_mode"
```

Comment:
If I read it correctly, after this all the e2e test will have this invalid resource test including adding an invalid resource and then delete it. I am wondering if there is a way to just run it with a single test without affecting others. General heath check is OK, but this applying bad one and delete can be a little bit confusing.

Reply:
This PR is not ready for review currently, just want to run pipeline workflows as a try. I will remove WIP once it is ready.
The tricky part is this test can't run parallel with other sub-tests in a workflow, so that it can only run before or after all the sub-tests. Thus, I add it before configuring broker credential, we need to configure the right credential after finishing this test anyway (BrokerCell is shared by broker related sub-tests).
From the pipeline test result, it seems four workflows will run auth check test (integration test for secret, integration test for wi, upgrade and conformance), we need this test for two integration workflows. Upgrade and conformance tests are sourcing the test_setup from `e2e-secret-tests.sh`, I will override their test-setup to exclude this test.

Reply:
Or probably use some workflow env to exclude this.